apollo-router/src/plugins/telemetry/formatters/mod.rs

```diff
@@ -91,14 +101,32 @@ where
     where
         W: std::fmt::Write,
     {
-        if (self.filter_fn)(event) {
+        if (self.filter_fn)(event) && self.rate_limit(event) {
             self.inner.format_event(ctx, writer, event)
         } else {
             Ok(())
         }
     }
 }
 
+impl<T, F> FilteringFormatter<T, F> {
+    fn rate_limit(&self, event: &tracing::Event<'_>) -> bool {
+        let now = Instant::now();
+        match self.last_logged.lock().entry(event.metadata().callsite()) {
+            std::collections::hash_map::Entry::Occupied(mut last) => {
+                if now - *last.get() < Duration::from_secs(10) {
+                    return false;
+                }
+                last.insert(now);
+            }
+            std::collections::hash_map::Entry::Vacant(entry) => {
+                entry.insert(now);
+            }
+        }
```

Comment:
Using callsite is the way to go I think.IIRC using entry is a less performant api for hashmap. I'm just worried about perfs here

Reply:
I could do a get() and insert() instead, it's not a big difference

Reply:
fixed in [346593a](https://github.com/apollographql/router/pull/4517/commits/346593ad690cb9db2bf86a6f1e9077fd88e77970)